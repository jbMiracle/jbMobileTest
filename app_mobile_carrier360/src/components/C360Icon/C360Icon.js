import React from 'react';
import map from 'lodash/map';
import PropTypes from 'prop-types';
import { createIconSetFromIcoMoon } from 'react-native-vector-icons';
import { connectStyle, IconNB, mapPropsToStyleNames } from 'native-base';
import icoMoonConfig from '../../assets/fonts/C360Icons/selection.json';

const names = map(icoMoonConfig.icons, 'properties.name');
const C360IconSet = createIconSetFromIcoMoon(icoMoonConfig);

C360IconSet.propTypes = {
  style: IconNB.propTypes.style,
  name: PropTypes.oneOf(names),
};

class C360IconNB extends IconNB {
  setIcon() {
    this.Icon = C360IconSet;
  }

  /**
   * Overwrite the render method of IconNB to provide acccurate proptype warnings
   *
   * @return {*}
   */
  render() {
    return (<C360IconSet
      ref={c => (this._root = c)} // eslint-disable-line
      { ...this.props }
    />);
  }
}

C360IconNB.propTypes = {
  style: IconNB.propTypes.style,
  name: PropTypes.oneOf(names),
};

export const StyledC360Icon = connectStyle(
  'NativeBase.IconNB',
  {},
  mapPropsToStyleNames
)(C360IconNB);

export default C360IconSet;
