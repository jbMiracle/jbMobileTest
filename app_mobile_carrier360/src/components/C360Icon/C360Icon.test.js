import * as _ from 'lodash';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { StyledC360Icon as Icon } from './C360Icon';

describe('C360Icon Component', () => {
  it('renders a result', () => {
    const rendered = TestRenderer.create(<Icon name='jbh-Hazmat' />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('renders a different result for different name key', () => {
    const optionPerk = TestRenderer.create(<Icon name='jbh-Hazmat' />);
    const optionFindLoads = TestRenderer.create(<Icon name='jbh-User_Group' />);

    expect(optionPerk.toJSON()).not.toEqual(optionFindLoads.toJSON());
  });

  it('responds to color attributes', () => {
    const colorNotSpecified = TestRenderer.create(<Icon name='jbh-User_Group' />);
    const colorSpecified = TestRenderer.create(<Icon
      color='#ff00ff'
      name='jbh-User_Group'
    />);

    const renderedColorPath = 'rendered.props.color';
    const noColorRendered = _.get(colorNotSpecified.toTree(), renderedColorPath);
    const colorRendered = _.get(colorSpecified.toTree(), renderedColorPath);

    expect(noColorRendered).not.toEqual(colorRendered);
  });
});
