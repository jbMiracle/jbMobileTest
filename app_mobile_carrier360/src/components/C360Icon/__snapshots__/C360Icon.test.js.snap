// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`C360Icon Component renders a result 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      Object {
        "color": "#000",
        "fontSize": 30,
      },
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
  virtual={undefined}
>
  Ó§ù
</Text>
`;
